quartz/components/scripts/search.inline.ts

```diff
@@ -64,13 +70,60 @@ function highlight(searchTerm: string, text: string, trim?: boolean) {
       }
       return tok
     })
+    .slice(startIndex, endIndex + 1)
     .join(" ")
 
   return `${startIndex === 0 ? "" : "..."}${slice}${
     endIndex === tokenizedText.length - 1 ? "" : "..."
   }`
 }
 
+function highlightHTML(searchTerm: string, el: HTMLElement) {
+  // try to highlight longest tokens first
+  const p = new DOMParser()
+  const tokenizedTerms = tokenizeTerm(searchTerm)
+  const html = p.parseFromString(el.innerHTML, "text/html")
+
+  const createHighlightSpan = (text: string) => {
+    const span = document.createElement("span")
+    span.className = "highlight"
+    span.textContent = text
+    return span
+  }
+
+  const highlightTextNodes = (node: Node) => {
+    if (node.nodeType === Node.TEXT_NODE) {
+      let nodeText = node.nodeValue || ""
+      tokenizedTerms.forEach((term) => {
+        const escapedTerm = term.replace(/[.*+?^${}()|[\]\\]/g, "\\$&") // Escape special characters
```

Comment:
why do we need to escape these?

Reply:
I was trying to escape latex.

Reply:
imo should be fine to not escape? what breaks if we dont do this

Reply:
nothing breaks actually.

Reply:
Hmm I don't remember why i did this (I wrote this at 5am td lol, just woke up).
<img width="974" alt="Screenshot 2024-02-01 at 16 32 33" src="https://github.com/jackyzha0/quartz/assets/29749331/1c185105-5669-41a6-9de2-a187d6cf3288">

but it seems like searching latex already works

and it doesn't break the latex node so I think we are good here, for now.

Reply:
awesome :) more code to axe

Reply:
already removed and finished.